includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1005,13 +1005,33 @@ private function get_disallowed_attributes_in_node( DOMElement $node, $attr_spec
 	 * Allowed values are found $this->globally_allowed_attributes and in parameter $attr_spec_list
 	 *
 	 * @param DOMElement $node                       Node.
+	 * @param array[][]  $tag_spec                   Tag spec list.
 	 * @param array[][]  $attr_spec_list             Attribute spec list.
 	 * @param DOMAttr[]  $attributes_pending_removal Attributes pending removal.
 	 * @return DOMAttr[]|false Attributes to remove, or false if the element itself should be removed.
 	 */
-	private function sanitize_disallowed_attribute_values_in_node( DOMElement $node, $attr_spec_list, $attributes_pending_removal ) {
+	private function sanitize_disallowed_attribute_values_in_node( DOMElement $node, $tag_spec, $attr_spec_list, $attributes_pending_removal ) {
 		$attrs_to_remove = [];
 
+		if (
+			! $node->hasAttribute( 'layout' ) &&
+			( $node->hasAttribute( 'width' ) || $node->hasAttribute( 'height' ) ) &&
+			isset( $tag_spec['amp_layout'] ) &&
+			! $this->has_valid_dimensions( $node )
```

Comment:
Perhaps the `has_valid_dimensions` method should take the `$node` and `$tag_spec` and perform all this logic.